### PR TITLE
Add BaseTracer#inject() method

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
And avoid using GlobalOpenTelemetry#getPropagators()

### CODE REVIEW

**Review by trask — APPROVED** [review]

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java [hunk 1]
```diff
@@ -253,6 +250,10 @@ public void addThrowable(Span span, Throwable throwable) {
     return propagators.getTextMapPropagator().extract(Context.root(), carrier, getter);
   }
 
+  public <C> void inject(Context context, C carrier, TextMapSetter<C> setter) {
```
Comment (jkwatson) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java:253]: public method in a public API should have some javadoc, I think.

**Review by jkwatson — APPROVED** [review]
would like some javadoc on the new method, but otherwise, 👍

**Review by anuraaga — APPROVED** [review]

File: instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientInstrumentation.java [hunk 2]
```diff
@@ -110,10 +110,7 @@ public HttpHost getTarget() {
     @Override
     public HttpRequest generateRequest() throws IOException, HttpException {
       HttpRequest request = delegate.generateRequest();
-      tracer()
-          .getPropagators()
-          .getTextMapPropagator()
-          .inject(context, request, tracer().getSetter());
+      tracer().inject(context, request, tracer().getSetter());
```
Comment (iNikem) [inline instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientInstrumentation.java:113]: This looks like an overload is missing.

**Review by iNikem — APPROVED** [review]

File: instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientInstrumentation.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/apache-httpasyncclient-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientInstrumentation.java:113]: Nice catch! Fixed.

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java:253]: Definitely. I'm planning to make a separate PR just to add Javadocs to `BaseTracer` sometime in the near future.

**Pull request merged**